src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -197,28 +203,44 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
                 this.logger.LogTrace("[RETURN_ATTEMPTED_HC_FAILED]");
                 return attempted;
             }
-
-            List<PeerAddress> notBanned = this.NotBanned().ToList();
-
-            int attemptedReachedThresholdCount = notBanned.Count(p => p.ConnectionAttempts == PeerAddress.AttemptThreshold);
-            bool areAllPeersReachedThreshold = attemptedReachedThresholdCount == notBanned.Count;
-            if (areAllPeersReachedThreshold)
-            {
-                this.logger.LogTrace("Resetting attempts for {0} addresses.", attemptedReachedThresholdCount);
-
-                // Reset attempts for all the peers since we've ran out of options.
-                foreach (PeerAddress peer in notBanned)
-                    peer.ResetAttempts();
-                
-                this.logger.LogTrace("(-)[RESET_ATTEMPTS]");
-                return notBanned;
-            }
+            
+            if (this.HasAllPeersReachedConnectionThreshold())
```

Comment:
@noescape00 after your changes, on first call to selected peers, it'll just reset the peers but still return the empty collection, then on second call it'll select peers as per normal. I think this is actually better? If you agree I'll approve 👍

Reply:
If you think it's better- let's keep it 🙂

Reply:
Otherwise I can quickly change it

Reply:
I think lets keep it like this, otherwise we are returning all the peers which isn't necessarily correct 👍